scipy/sparse/csgraph/_shortest_path.pyx

```diff
@@ -68,7 +70,7 @@ def shortest_path(csgraph, method='auto',
                      Computational cost is approximately ``O[N^3]``.
                      The input csgraph will be converted to a dense representation.
 
-           'D'    -- Dijkstra's algorithm with Fibonacci heaps.
+           'D'    -- Dijkstra's algorithm with priority queue.
                      Computational cost is approximately ``O[N(N*k + N*log(N))]``,
```

Comment:
It appears the complexity currently stated is wrong even before the change. Dijkstra's worst complexity without a priority queue is `O(N^2)`. So the current complexity is definitely not `O(N^2 * log(N))`
The correct complexity should be `O[(N*k + N)*log(N)]` (assuming `N*k` is a logical way to compute the number of edges in the graph).
Should we state the [known](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm#Running_time) complexity of `O[(E + N)*log(N)]` where `E` is the number of edges?

Reply:
@tsery-ns I think that the complexity currently stated is actually correct if `indices=None`. On the wikipedia page that you linked it is stated that Dijkstra's algorithm with Fibonacci heap has a complexity of $\Theta(|E| + |V| \log|V|)$ where $|E|$ is the number of edges and $|V|$ is the number of vertices.
Since the `shortest_path` function finds the shortest path for all the vertex pairs in the graph (like the Floyd-Warshall algorithm does), the Dijkstra's algorithm is run for each vertex of the graph leading to a complexity of $O[|V|(|E| + |V| \log|V|)]$.

Instead, if my reasoning is correct, in the case `indices!=None`, if we denote as $|I|$ the length of `indices`, the complexity should be $O[|I|(|E| + |V| \log|V|)]$ because we need to run Dijkstra's algorithm only $|I|$ times.

Reply:
@tsery-ns @Kai-Striega As this pull request seems to be moving forward, I
would like to reiterate my previous comment. The proposed change to the stated
computational cost may be inaccurate and potentially misleading.

Reply:
@tsery-ns I think this needs to be addressed before we merge.

Reply:
Rephrasing to 
```

                     Computational cost is approximately ``O[I*(E + N*log(N))]``,
                     where ``E`` is the number of edges in the graph,
                     and ``I = len(indices)`` if ``indices`` is passed. Otherwise,
                     ``I = N``.
```

Reply:
@Giuseppe499 does that look correct to you?

Reply:
It looks right to me if the priority queue is implemented using a Fibonacci heap. If instead, for example, the priority queue is implemented with a binary heap, the computational cost should be `O[I*(E + N)*log(N)]`. I'm not sure how std::priority_queue is implemented and if it is system dependent. On "The C++ Programming Language (4th Edition)" by Bjarne Stroustrup it says 
> A `priority_queue` is almost certainly implemented using a `heap`.

Therefore, I think that `O[I*(E + N)*log(N)]` is the safest guess.
For reference, my reasoning on the computational costs is based on [this](https://en.wikipedia.org/wiki/Dijkstra%27s_algorithm#Running_time).

Reply:
According to [GeeksForGeeks](https://www.geeksforgeeks.org/priority-queue-in-cpp-stl/):
`In C++ STL, the top element is always the greatest by default. We can also change it to the smallest element at the top. Priority queues are built on the top of the max heap and use an array or vector as an internal structure. In simple terms, STL Priority Queue is the implementation of [Heap Data Structure](https://www.geeksforgeeks.org/heap-data-structure/).`

What should be the complexity in this case?